src/server/htmlServer.ts

```diff
@@ -184,14 +205,14 @@ function getEditedLiquidExpression(colIndex: number, editedLine: string) {
             liquidForAutocomplete.AutoCompleteAtIndex = colIndex - startIndexOfEditedLiquidExpression;
         }
     } catch (e) {
-        // Add Telemetry for index out of bounds...not a proper liquid expression
+        // Add Telemetry for index out of bounds...not a proper liquid expression. This may again bloat up the logs (since the autocomplete events can be fired even for non-portal html files)
```

Comment:
intersting, in general we should prefer to not use exceptions to handle control flow. This isn't part of your PR for telemetry, but the comment seems to indicate the code block is missing a guard before indexing. Catching all exceptions and swallowing is also a bad pattern. There may be a lot of places where runtime code is failing that we'll never get any error report.

Reply:
Appreciate the idea of global exception handler. Will work on this in the next PR.